docs/src/testing/testing-contract-internals.md

```diff
@@ -1,31 +1,45 @@
 # Testing Contracts' Internals
 
-Sometimes, you want to test a function which uses Starknet context (like block number, timestamp, storage access)
-without deploying the actual contract.
+Sometimes, you want to test a function that uses Starknet context such as block number, timestamp or storage access.
 
-Since every test is treated like a contract, using the aforementioned pattern you can test:
+Since every test is treated like a contract (with the address [`test_address`](#snforge_stdtest_address---address-of-test-contract)), you can use the aforementioned pattern to test:
 - functions which are not available through the interface (but your contract uses them)
 - functions which are internal
 - functions performing specific operations on the contracts' storage or context data
 - library calls directly in the tests
 
 ## Utilities For Testing Internals
-To facilitate such use cases, we have a handful of utilities which make a test behave like a contract.
 
 ### `contract_state_for_testing()` - State of Test Contract
 
-This is a function generated by the `#[starknet::contract]` macro.
-It can be used to test some functions which accept the state as an argument, see the example below:
+This function is generated by the `#[starknet::contract]` macro.
+It can be used to test functions that accept the state as an argument.
+You can either use it directly without deploying a contract or use it to modify the state of an existing contract.
+
+> ⚠️ **Warning**
+>
+> When using `contract_state_for_testing` without the `interact_with_state` cheatcode, the storage is modified in the context of the [`test_address`](#snforge_stdtest_address---address-of-test-contract) contract.
+
+#### Testing state without a deployed contract
 
 ```rust
 {{#include ../../listings/testing_contract_internals/src/basic_example.cairo}}
 ```
 
 This code contains some caveats:
 1. To access `read/write` methods of the state fields (in this case it's `balance`) you need to also import `StoragePointerReadAccess` and `StoragePointerWriteAccess` for reading and writing respectively.
-2. To access functions implemented directly on the state you need to also import an appropriate trait or function.
+2. To access functions implemented directly on the state, you need to also import an appropriate trait or function.
 3. This function will always return the struct keeping track of the state of the test. It means that within one test every result of `contract_state_for_testing` actually points to the same state.
 
+#### Modifying the state of an existing contract
+
+There is a special [`interact_with_state`](../appendix/cheatcodes/interact_with_state.md) cheatcode dedicated for using `contract_state_for_testing` with a deployed contract.
+By using this function, the state will be modified for the provided contract address.
+
+```rust
+{{#include ../../listings/testing_contract_internals/src/using_interact_with_state.cairo}}
+```
```

Comment:
We are importing the whole big snippet here, is that good for the readability of the docs? Maybe we should at least split it by `#[test]` functions used in the snippet?